### PR TITLE
JAPI-79 Throw exception if compiled state not reached in wucreateandupdate

### PR DESCRIPTION
**Pull request by Michael-Gardner**
Signed-off-by: Michael Gardner <michael.gardner@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [x] I have created a corresponding JIRA ticket for this submission
- [x] My code follows the code style of this project.
  - [ ] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [ ] The change has been fully tested:
  - [ ] I have performed unit tests to cover my changes.
  - [ ] I have performed system test and covered possible regressions and side effects.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.

### CODE REVIEW

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsWorkUnitsClient.java [hunk 1]
```diff
@@ -1388,27 +1388,35 @@ public WorkunitInfo createWUFromECL(WorkunitInfo wu) throws Exception
...
-
-            }
-            else
+            if (exceptions != null)
+        	    throwWsWUExceptions(exceptions, "Error compiling ECL query");
+            if (wuUpdateResponse.getWorkunit().getStateID() != 1) // not immediately in compiled state
             {
-                throwWsWUExceptions(exceptions, "Error compiling ECL query");
+                WUWaitCompiled wuwaitcompiledparameters = new WUWaitCompiled();
+                wuwaitcompiledparameters.setWait(10); // seconds
```
Comment (rpastrana) [inline wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsWorkUnitsClient.java]: these timeouts are always so tough to get right, does it make sense to let the caller pass in this value?

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsWorkUnitsClient.java [hunk 2]
```diff
@@ -1388,27 +1388,35 @@ public WorkunitInfo createWUFromECL(WorkunitInfo wu) throws Exception
...
-                createdWU.setCluster(wu.getCluster());
-                createdWU.setJobname(wu.getJobname());
-                createdWU.setSleepMillis(wu.getSleepMillis());
-                createdWU.setOriginalEclWatchUrl(getEclWatchUrl());
-
-            }
-            else
+            if (exceptions != null)
+        	    throwWsWUExceptions(exceptions, "Error compiling ECL query");
+            if (wuUpdateResponse.getWorkunit().getStateID() != 1) // not immediately in compiled state
```
[the remark this answers is not in the document]
Comment (drealeed) [reply wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsWorkUnitsClient.java]: Does it make sense to use wu.maxMonitorMillis? or the defaultWaitTime variable?

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsWorkUnitsClient.java (hunk 1 above)
Comment (drealeed) [reply wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsWorkUnitsClient.java]: Does it make sense to use wu.maxMonitorMillis? or the defaultWaitTime variable?

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsWorkUnitsClient.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (Michael-Gardner) [reply wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsWorkUnitsClient.java]: Yes it does Drea,  I'll add that in this morning

**Comment by Michael-Gardner** [comment]
@drealeed now we take the maxMonitorMillis variable (converted into seconds) for the wait time.

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsWorkUnitsClient.java [hunk 3]
```diff
@@ -1388,27 +1388,35 @@ public WorkunitInfo createWUFromECL(WorkunitInfo wu) throws Exception
...
             {
-                throwWsWUExceptions(exceptions, "Error compiling ECL query");
+                WUWaitCompiled wuwaitcompiledparameters = new WUWaitCompiled();
+                wuwaitcompiledparameters.setWait((wu.getMaxMonitorMillis()/100)); // seconds
+                wuwaitcompiledparameters.setWuid(wuUpdateResponse.getWorkunit().getWuid());
+                WUWaitResponse wuWaitResponse = wsWorkunitsServiceSoapProxy.WUWaitCompiled(wuwaitcompiledparameters);
+                exceptions = wuWaitResponse.getExceptions();
+                if (exceptions != null)
+                    throwWsWUExceptions(exceptions, "Error compiling ECL query");
+                if (wuWaitResponse.getStateID() != 1) // not in compiled state
```
Comment (rpastrana) [inline wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsWorkUnitsClient.java]: WUState.COMPILED ?

Comment (rpastrana) [reply wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsWorkUnitsClient.java]: @Michael-Gardner bump

**Pull request closed without merging**